rfcs/0182-user-management.md

```diff
@@ -0,0 +1,306 @@
+  Code duplications can be refactored as they appear.
+
+  Indeed it is way easier to deal with code duplication in a monorepo: since the
+  barrier of communication between multiple repositories disappears, a
+  duplicated code can be factored more easily, requiring just one pull request
+  instead of one per repo.
+
+- Evaluation times of monorepo will increase.
+
+  The increase in evaluation time is a bit harder to measure. However,
+
+  - "increase in evaluation time" was not an argument strong enough for undoing
+    the NixOS assimilation;
+  - arguably, the increase in evaluation time will be felt by those effectively
+    using the hometool; the remaining users will not be signficatively affected.
+  - a similar argument for the increase in size can be crafted here.
+
+  The perceived advantages of having this module system surpasses these small
+  disadvantages
+
+# Alternatives
+[alternatives]: #alternatives
+
+- The trivial "do nothing".
+
+- Promote an alternative toolset.
+
+  The most viable alternative in this field is home-manager. However it is not
+  so easy to assimilate a consolidated project within another consolidated
+  project, not only in terms of source code, but also in terms of community and
+  decision-making.
+
+  A better approach would be promote some form of a healthy competition, in
+  which home-manager and hometool are different projects competing for the same
+  niche, each with their own teams, communities, design projects etc.; further,
+  both teams are free to share code between them, provided they follow their
+  corresponding licenses in doing so.
+
+- Promote `guix home`
+
+  What? Lisp is cool!
+
+# Prior art
+[prior-art]: #prior-art
+
+## Guix Home
+
+As an example of prior art, there is our Scheme-based cousin, Guix Software
+Distribution. Since at least 2022 AD they bring a similar tool, conveniently
+called Guix Home.
+
+The nicest thing about this tool is that it is tightly integrated with Guix, to
+the point of `home` being a mere subcommand of `guix`.
+
+## Home Manager
+
+Home Manager is a well-established toolset that leverages the Nixpkgs module
+system to allow basic users to manage their user-specific environment in a
+declarative way.
+
+## Wrapper Manager
+
+Wrapper-manager is a Nix library that allows the user to configure your favorite
+applications without adding files into `~/.config`. This is done by creating
+wrapper scripts that set the appropriate environment set - variables, flags,
+configuration files etc. - to the wrapped program.
+
+# Unresolved questions
+[unresolved]: #unresolved-questions
+
+Given that this tool will live inside Nixpkgs monorepo, it is expected that
+future packages will interact with this new tool. How those interactions should
+be dealt?
+
+# Future work
+[future]: #future-work
+
+- Update an extend the CI
+- Set expectations on portability among present and future platforms Nixpkgs
+  supports
+  - Especially outside NixOS
+  - Especially outside Linux
```

Comment:
I disagree. Home management would be a part of the package manager, not NixOS. I don't see why derivations would work on other systems but home management won't, it feels like a regression to me.

Also, #163 should pave way for the implementation of things detailed in this RFC if/when accepted, making support for other platforms not so out of scope, IMO.

Reply:
> Home management would be a part of the package manager, not NixOS. I don't see why derivations would work on other systems but home management won't, it feels like a regression to me.

My interpretation of this was moreso that it was encouraging better support for platforms in derivations, not this tool specifically. If this *is* speaking about `hometool`, I'm still not sure why it would be considered "future work". Portability should be considered from the beginning here, as frankly there isn't much reason to make most modules platform specific; for example in home-manager, a vast majority of modules are already only limited by the package's platform support

If I was correct in my initial reading though, I think my point still stands. Allowing derivations to build for more targets doesn't have much to do with introducing ways to configure them

Reply:
Well, if you're correct on that, then nixpkgs already supports more targets even if not formally listed in some policy.

If it's regarding `hometool`, then yes it should be done from the get-go and not defer it for later.

Reply:
Well, in my experience different distros do not share `/home` well.

Reply:
> Well, in my experience different distros do not share /home well.

I'm a bit confused what this has to do with the current thread. I don't believe sharing `/home` was mentioned anywhere

Reply:
I this the wording here is ambiguous. @AndersonTorres by "platform" do you mean architectures or OS/distributions

Reply:
> I'm a bit confused what this has to do with the current thread. I don't believe sharing `/home` was mentioned anywhere

Anecdotal evidence ahead.

In the old times I was switching from Slackware to Arch, my `/home` lived in a separate partition.
When I installed a second distro, the programs in the second distro did not recognize what was already present in the `/home`. The distros did not share `/home` well.

Reply:
> In the old times I was switching from Slackware to Arch, my /home lived in a separate partition.
When I installed a second distro, the programs in the second distro did not recognize what was already present in the /home. The distros did not share /home well.

I'm well aware of these types of issues, but again I'm not sure what it has to do with the current thread. Are you implying this tool should act similar to sharing a /home but not have the typical issues associated with it?

> Operating system - and everything it implies but we usually abstract away (libc, userland etc.).

This needs to be specified in the RFC. The current wording is (once again) vague